storage/namespace/metadata.go

```diff
@@ -33,16 +35,31 @@ type metadata struct {
 // NewMetadata creates a new namespace metadata
 func NewMetadata(id ts.ID, opts Options) Metadata {
 	copiedID := checked.NewBytes(append([]byte(nil), id.Data().Get()...), nil)
-	return metadata{
+	return &metadata{
```

Comment:
Lots of places call `Metadata.Options()` to retrieve the retention. Made it a pointer to avoid the cost of copying underlying struct for each method call.